_includes/articles/bitgo-musig2.md

```diff
@@ -0,0 +1,186 @@
+{:.post-meta}
+*by [Brandon Black][] from [BitGo][]*
+
+The first [MuSig paper][] was published in 2018, and the potential of
+[MuSig][topic musig] on
+Bitcoin was one of the selling points used to gain support for the taproot soft
+fork. Work on MuSig continued with the publication of [MuSig-DN][] and [MuSig2][] in 2020.
+When taproot neared activation on bitcoin’s mainnet in 2021, excitement
+about bringing MuSig signing to bitcoin users was palpable. At BitGo, we were
+hoping to launch a MuSig taproot wallet concurrent with taproot activation; but
+the spec, test vectors, and reference implementation were incomplete. Instead,
+BitGo [launched][bitgo blog taproot] the first tapscript multisig wallet and made the [first tapscript
+multisig transaction][] on mainnet. Nearly two years later, MuSig2 is specified in
+[BIP327][], and we [launched][bitgo blog musig2] the first MuSig taproot multisig wallet.
+
+## Why MuSig2?
+
+### Compared to Script Multisig
+
+There are two main
+benefits of MuSig compared to script multisig. The first and most obvious
+benefit is a reduced transaction size and miner fee. On-chain signatures are
+64-73 bytes, 16-18.25 virtual bytes (vB), and MuSig can combine two (or more)
+signatures into one. In BitGo’s 2-of-3 case, a MuSig key path input costs
+57.5vB, compared to a native SegWit input at 104.5vB or depth 1
+[tapscript][topic tapscript] at
+107.5vB. The second benefit of MuSig is an improvement in privacy. With a MuSig
+key path on a collaboratively held output, a cooperative spend cannot be
+distinguished by a third party blockchain observer from a single signature
+taproot spend.
+
+Naturally, there are some drawbacks to MuSig2. Two important ones revolve around
+[nonces](#nonces-deterministic-and-random). Unlike signers for plain ECDSA (elliptic curve digital signature
+algorithm) or [Schnorr signatures][topic schnorr signatures], MuSig2 signers cannot consistently use
+deterministic nonces. This inability makes it more difficult to ensure
+high-quality nonces and to ensure against nonce reuse. MuSig2 requires two
+rounds of communication in most cases. First nonce exchange, and then signing.
+In some cases, the first round can be precomputed, but this must be undertaken
+cautiously.
+
+### Compared to Other MPC protocols
+
+MPC (multi-party compute) signing protocols are growing in popularity due to the
+aforementioned fee and privacy benefits. MuSig is a _simple_ multi-signature
+(n-of-n) protocol, made possible by Schnorr signatures’ linearity. MuSig2 can be
+explained in a 30-minute presentation, and the complete reference implementation
+is 461 textual lines of Python. [Threshold signature][topic threshold signature] (t-of-n) protocols, such as
```

Comment:
“Textual lines” throws me off here. It’s lines of code, right?

```suggestion
is 461 lines of code in Python. [Threshold signature][topic threshold signature] (t-of-n) protocols, such as
```

Reply:
I used textual lines because people use many different definitions for "lines of code" are they non-empty lines? non-empty, non-comment lines? 🤷 It's 461 lines in a text editor from top to bottom of the reference implementation.